docs/source/control_tutorials.rst

```diff
@@ -355,26 +364,169 @@ As with :py:class:`.MachineController`, :py:class:`.BMPController` supports the
+
+As an example, to send an SCP packet core 1 on chip (0, 0) with a ``cmd_rc`` of
+``123``::
+
+    >>> from rig.machine_control.packets import SCPPacket
+    >>> data = b"Hello world!\0"
+    >>> packet = SCPPacket(
+    ...     dest_port=1,
+    ...     dest_x=0, dest_y=0, dest_cpu=1,
+    ...     cmd_rc=123
+    ...     data=data
+    ... )
+    >>> out_sock.send(packet.bytestring)
+
+On the receiving core, the ``on_sdp_from_host`` callback might then look like
+this:
+
+.. code-block:: c
+
+    void on_sdp_from_host(uint mailbox, uint port)
+    {
+      sdp_msg_t *msg = (sdp_msg_t *)mailbox;
+      if (msg->cmd_rc == 123)
+      {
+        io_printf(IO_BUF,
+                  "Got SCP packet from host with data: %s\n",
+                  msg->data);
+      }
+      spin1_msg_free(msg);
+    }
 
 .. note::
-    Since different applications typically have very different requirements for
-    SDP and SCP support, Rig does not currently offer any high-level support for
-    their use. The developers are open to discussions about potential
-    (appropriate) high-level interfaces.
+
+    SpiNNaker can only recieve packets up to a certain size. This size can be
+    determined using :py:class:`~rig.machine_control.MachineController`'s
+    :py:meth:`~rig.machine_control.MachineController.scp_data_length` property
+    This property defines the maximum length of the data-field in an SCP packet
+    sent to the machine.
+
+
+Example: Receiving SDP packets from a running application
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To receive SDP packets from an application first a socket must be opened
+through which the SDP packets may be received. A port number of your choosing
+may be used for this purpose. For example::
+
+    >>> import socket
+    >>> PORT = 50007
+    >>> in_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    >>> in_sock.bind(("", PORT))
+
+Next, you must set up an 'IP tag' on every Ethernet-connected SpiNNaker chip
+through which SDP packets may be sent back to the host which informs SpiNNaker
+of the IP address these packets should be sent to.
+
+A list of the Ethernet-connected chips in a typical SpiNNaker machine can be
+produced using :py:class:`rig.machine_control.MachineController.get_machine`
+and :py:class:`rig.geometry.spinn5_eth_coords` and an IP tag configured on each
+using :py:class:`rig.machine_control.MachineController.iptag_set` like so::
+
+    >>> from rig.machine_control import MachineController
+    >>> from rig.geometry import spinn5_eth_coords
+    
+    >>> # Get the IP and port of the socket we opened
+    >>> addr, port = in_sock.getsockname()
+    
+    >>> # Set-up IP Tag 1 on each ethernet-connected chip to forward all SDP
+    >>> # packets to this socket.
+    >>> mc = MachineController("spinnaker-machine-hostname")
+    >>> machine = mc.get_machine()
+    >>> for x, y in spinn5_eth_coords(machine.width, machine.height):
+    ...     mc.iptag_set(1, addr, port, x, y)
```

Comment:
...true... When the pinus-rigida code-review happens I suggest we have a major overview of things which have appeared in all our applications and see where these common things live and if any should be put into rig. This may be one such thing...